File: DSA/Shreya_Goyal_590015491/Day_19_Question_2.c
#include <stdio.h>

int main() {
    int arr[7], n = 7, i;
    int left, right, mid;
    
    printf("Enter 7 numbers:\n");
    for(i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    left = 0;
    right = n - 1;

    while(left <= right) {
        mid = (left + right) / 2;

        if ((mid == 0 || arr[mid] >= arr[mid - 1]) && 
            (mid == n - 1 || arr[mid] >= arr[mid + 1])) {
            printf("Peak: %d\n", mid);
            break;
        }
        else if (mid > 0 && arr[mid - 1] > arr[mid]) {
            right = mid - 1;
        }
        else {
            left = mid + 1;
        }
    }

    return 0;
}

File: DSA/VaniAgrawal_590015846/Day_19_Question_2.py
# Find any peak element's index using efficient divide-and-conquer approach.
arr = list(map(int, input("Enter the array elements separated by space: ").split()))
n = len(arr)

low = 0
high = n - 1
peak_index = -1

while low <= high:
    mid = (low + high) // 2

    left = (mid == 0 or arr[mid] > arr[mid - 1])
    right = (mid == n - 1 or arr[mid] > arr[mid + 1])

    if left and right:
        peak_index = mid
        break
    elif mid > 0 and arr[mid - 1] > arr[mid]:
        high = mid - 1
    else:
        low = mid + 1

print(peak_index)
